### PR TITLE
Add method to create SupervisionSet from RTTM files

### PR DESCRIPTION
**Pull request by desh2608**
I'm not sure how useful this method would be for others (I have been using this for some of my own work). If you think it can be added, I will include some tests.

### CODE REVIEW

File: lhotse/supervision.py [hunk 1]
```diff
@@ -512,6 +512,50 @@ def from_dicts(data: Iterable[Dict]) -> "SupervisionSet":
             SupervisionSegment.from_dict(s) for s in data
         )
 
+    @staticmethod
+    def from_rttm(path: Pathlike) -> "SupervisionSet":
+        """
+        Read an RTTM file located at ``path`` and create a :class:`.SupervisionSet` manifest for them.
+        Can be used to create supervisions from custom RTTM files (see, for example, :class:`lhotse.dataset.DiarizationDataset`).
+
+        Use the :meth:`.from_rttms` method if you want to read multiple RTTM files at once.
```
Comment (pzelasko) [inline lhotse/supervision.py:521]: How about modifying this function to accept both a single Pathlike and an iterable of them (you can check with `isinstance`)?

File: lhotse/supervision.py [hunk 2]
```diff
@@ -512,6 +512,50 @@ def from_dicts(data: Iterable[Dict]) -> "SupervisionSet":
             SupervisionSegment.from_dict(s) for s in data
         )
 
+    @staticmethod
+    def from_rttm(path: Pathlike) -> "SupervisionSet":
```
Comment (pzelasko) [inline lhotse/supervision.py:516]: It would be nice to add a short description of RTTM format in this docstring (which column stands for what)

**Review by pzelasko — COMMENTED** [review]
Looks good!

**Comment by desh2608** [comment]
@pzelasko done. Also added a simple test.

**Comment by desh2608** [comment]
Do not merge this yet --- I am looking for a way to make the fixture as a tmpfile from the code itself instead of adding it in the fixtures directory.

File: lhotse/supervision.py [hunk 3]
```diff
@@ -512,6 +513,89 @@ def from_dicts(data: Iterable[Dict]) -> "SupervisionSet":
...
+                        channel=int(parts[2]),
+                        start=float(parts[3]),
+                        duration=float(parts[4]),
+                        speaker=parts[7],
+                    )
+                )
+        return SupervisionSet.from_segments(segments)
+
+    @staticmethod
+    def from_rttms(paths: Union[Pathlike, Iterable[Pathlike]]) -> "SupervisionSet":
```
Comment (pzelasko) [inline lhotse/supervision.py:574]: I meant if we can merge `from_rttms` and `from_rttm` into a single method that auto-detects the right thing to do with a single path vs an iterable of them. The names differ by just one character which makes them easy to confuse. WDYT?

File: lhotse/supervision.py [hunk 4]
```diff
@@ -512,6 +513,89 @@ def from_dicts(data: Iterable[Dict]) -> "SupervisionSet":
...
+
+        .. hint::
+
+            For creating a SupervisionSet from a single RTTM file, use the :meth:`.from_rttm` method.
+
+        :param paths: Path to RTTM files (can be path to a directory or an iterable of paths to RTTM files).
+        :return: a new ``SupervisionSet`` instance containing segments from the RTTM files.
+        """
+        if isinstance(paths, Pathlike.__args__):
+            paths = check_and_rglob(paths, "*.rttm")
```
Comment (pzelasko) [inline lhotse/supervision.py:592]: I think this rglob pattern wouldn't find files named `rttm` which are somewhat common in Kaldi recipes?

File: lhotse/supervision.py (hunk 3 above)
Comment (desh2608) [reply lhotse/supervision.py:574]: Yeah that makes sense. I'll make that change.

File: lhotse/supervision.py (hunk 4 above)
Comment (desh2608) [reply lhotse/supervision.py:592]: I suppose in that case I will remove rglob and just assume that the user has provided the path to a file or an iterable of paths.

**Comment by desh2608** [comment]
Maybe ready to merge? @pzelasko

**Comment by pzelasko** [comment]
Looks great, thanks!

**Pull request merged**